### ISSUE TITLE
False negative with callable type compatibility involving positional-only parameters

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6820]
Pyright doesn't catch the following type compatibility issue:

```python
class Proto11(Protocol):
    def __call__(self, x: int, /, y: str) -> Any:
        ...

def cb11_good1(x: int, /, y: str, z: None = None) -> Any:
    ...

def cb11_good2(x: int, y: str, z: None = None) -> Any:
    ...

def cb11_bad1(x: int, y: str, /) -> Any:
    ...

cb11: Proto11 = cb11_good1  # OK
cb11 = cb11_good2  # OK
cb11 = cb11_bad1  # Type error: y is position-only
```

See test case callbackProtocol7.py.

**Comment by erictraut** [comment]
This is a duplicate of #6833.
